### ISSUE TITLE
Alternative dependencies not recognized

### ISSUE THREAD

**Issue by 0** [Bcfg2/bcfg2#386]
Debian has support for alternative dependencies for a package, so that the presence of any single one is sufficient. For example, `bcfg2-server` depends on `python-pyinotify | python-gamin`. If neither is already present, `apt` installs one.

The problem is that bcfg2 sees the installed dependency as an extra entry and tries to remove it.

My current workaround is to explicitly add one of the alternative dependencies to the bundle.

**Comment by AlexanderS** [comment]
That is expected. The bcfg2-server does not know if some package is installed on the server. Currently or-dependencies are simply ignored. AFAIK apt tries to install always the first package (if it has no other information), so the server could try to search all specified packages (and dependencies) for one of the dependencies and simply pick the first one if nothing is found. But this behavior currently does not exist.

**Comment by solj** [comment]
Yeah, this is expected behavior. You have to explicitly tell the server which dependency you want because we don't want to make that choice for you. Perhaps we should document this behavior?
